components/automate-ui/src/app/entities/notification_rules/notification_rule.model.ts

```diff
@@ -67,13 +79,17 @@ export class NotificationRule implements RuleInterface {
     return Object.keys(this.AlertTypeLabels);
   }
 
+  public getTargetTypeKeys(): string[] {
+    return Object.keys(this.TargetTypeLabels);
+  }
+
   public toRequest(): Object {
```

Comment:
Still need to fix this:

> "Then toRequest should return the union type `NotificationTarget | NotificationRule`" 

That is, the type should be in the function signature not at the call site:
And then this changes, too:
https://github.com/chef/automate/blob/3435b2f10d26bd341724df0e024ba4ff7005282f/components/automate-ui/src/app/entities/notification_rules/notification_rule.requests.ts#L61

So that changes to:
```
const response = rule.toRequest();
```

Also, same change here in rule.ts:
https://github.com/chef/automate/blob/da403c721a5576b4f28316daa2f52492bfaade04/components/automate-ui/src/app/pages/notifications/rule.ts#L80

and here: https://github.com/chef/automate/blob/740683eee0890a7f960b5787ba31979d505e01e2/components/automate-ui/src/app/services/rules/rules.service.ts#L73

Reply:
@msorens 
rule.ts is going to remove in my next PR, because we are shifted the most of code in notification_rule.ts and next PR is to create a notification using modal.

regarding `public toRequest()` function, I tried to define the union type like 
```
public toRequest(): (NotificationTarget | NotificationRule) {
```
but is not working I am getting like 
![Screenshot_from_2020_06_23_18_22_56](https://user-images.githubusercontent.com/12297653/85407532-f4757080-b580-11ea-8d96-084cc6332e9b.png)

also in 
![Screenshot_from_2020_06_23_18_41_48](https://user-images.githubusercontent.com/12297653/85407909-76659980-b581-11ea-9477-7317a7d02c64.png)

Reply:
Let's back up a bit. What does toRequest() actually return? It is an object with a `rule` property whose value is itself an object with 3 properties. But it is peculiar that one of those 3 properties has a variable name. Couldn't we keep property names constant?
```
      'rule': {
        'name': this.name,
        'event': this.ruleType,
        'targetType': this.targetType,
        'target': target
      }
```
Then, the return type of `toRequest` will just be the type of this object, so you want to define a new interface that specifies this type.